public/app/features/alerting/unified/components/rule-editor/notificaton-preview/useGetAlertManagersSourceNamesAndImage.tsx

```diff
@@ -12,10 +11,13 @@ export const useGetAlertManagersSourceNamesAndImage = () => {
   //get current alerting config
   const { currentData: amConfigStatus } = alertmanagerApi.useGetAlertmanagerChoiceStatusQuery(undefined);
 
-  const externalDsAlertManagers: AlertManagerNameWithImage[] = useExternalDataSourceAlertmanagers().map((ds) => ({
-    name: ds.dataSource.name,
-    img: ds.dataSource.meta.info.logos.small,
-  }));
+  const externalDsAlertManagers = getAlertManagerDataSources()
```

Comment:
Nice!

Reply:
Can we add a function to `public/app/features/alerting/unified/utils/datasource.ts` and call it here? We might be able to use it in other places in the future if we want to circumvent the API call to `//api/v1/ngalert/alertmanagers`